plugins/tplink/tl_sg108e

```diff
@@ -0,0 +1,210 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# $Id$
+#
+# Magic markers (used by munin-node-configure and some installation scripts):
+#%# family=auto
+#%# capabilities=autoconf
+
+use strict;
+use warnings;
+use WWW::Mechanize;
+
+=pod
+
+=encoding UTF-8
+
+=head1 NAME
+
+tl_sg108e - Plugin to monitor packets per second  and link speed for TP-Link SG108E switches 
+
+=head1 APPLICABLE SYSTEMS
+
+TP-Link SG108E switches with web management (http). Tested with software version 1.0.2 Build 20160526 Rel.34615
+
+=head1 CONFIGURATION
+
+Add this to the relevant munin-node config file. You can specify switch address, username, password and description for each port
+(the switch management doesn't allow port descriptions)
+
+  [tl_sg108e]
+  env.switch 192.168.1.12
+  env.username admin
+  env.password mySecretPassword
+  env.p1 'Link to PC1'
+  env.p2 'Link to server1'
+  env.p3 'Not used'
+  env.p4 'Link to AP'
+  env.p5 'Link to PC2'
+  env.p6 'Link to PC3'
+  env.p7 'Not used'
+  env.p8 'Uplink'
+
+Requires WWW:Mechanize module:
+  sudo apt-get install libwww-mechanize-perl
+
+=head1 BUGS/GOTCHAS
+
+The link speed is represented as a number:
+  0 - down
+  5 - 100Mbps full
+  6 - 1Gbps
+
+=head1 AUTHOR
+
+Adrian Popa (https://github.com/mad-ady)
+
+=head1 COPYRIGHT
+
+Copyright (c) 2018, Adrian Popa
+
+All rights reserved. This program is free software; you can
+redistribute it and/or modify it under the terms of the GNU General
+Public License as published by the Free Software Foundation; version 2
+dated June, 1991.
+
+=head1 VERSION
+
+  $Id$
+
+=cut
+
+# read parameters from munin
+my $switch   = ( $ENV{switch}   || '192.168.1.1' );
+my $username = ( $ENV{username} || 'admin' );
+my $password = ( $ENV{password} || 'admin' );
+my $numberOfPorts = 8;
+
+#populate the ports and descriptions based on ENV
+my %ports = ();
+for ( 1 .. $numberOfPorts ) {
+    my $i = $_;
+    if ( defined $ENV{"p$i"} ) {
+        $ports{$i} = $ENV{"p$i"};
+    }
+    else {
+        #no description
+        $ports{$i} = "";
+    }
+}
+
+if ( $ARGV[0] and $ARGV[0] eq 'autoconf' ) {
+    print "no (manual configuration needed)\n";
+    exit 0;
+}
+
+if ( $ARGV[0] and $ARGV[0] eq "config" ) {
+    foreach my $graphType (qw/Packets Speed/) {
+        foreach my $port ( sort keys %ports ) {
+            print "multigraph ${graphType}_if_$port\n",
+              "graph_title $graphType for $switch port $port $ports{$port}\n",
+              "graph_info $graphType graph for port $port $ports{$port}\n",
+              "graph_args --base 1000 -l 0\n",
+              "graph_scale yes\n",
+              "graph_category switch\n";
```

Comment:
Please use `network` instead.
Or pick another suitable category from [the list of well-known categories](http://guide.munin-monitoring.org/en/latest/reference/graph-category.html#well-known-categories).

Reply:
Thanks, I didn't know about that. Will change category. I'd also like to change and have the switch display as a standalone "remote" device, but I haven't gotten it to work yet.